src/ui/DynamicFacet/DynamicFacetValues/DynamicFacetRangeValueUtils.ts

```diff
@@ -1,50 +1,73 @@
+import * as Globalize from 'globalize';
 import { RangeType, IRangeValue, RangeEndScope } from '../../../rest/RangeValue';
 import { DynamicFacetRangeValueFormat } from '../DynamicFacetRange';
 import { NumberUtils } from '../../../utils/NumberUtils';
-import * as Globalize from 'globalize';
 import { isNull } from 'util';
+import { DateUtils } from '../../../utils/DateUtils';
 
 export class DynamicFacetRangeValueUtils {
-  public static formatValue(value: RangeType, valueFormat: DynamicFacetRangeValueFormat) {
+  public static formatDisplayValue(value: RangeType, valueFormat: DynamicFacetRangeValueFormat) {
     switch (valueFormat) {
       case DynamicFacetRangeValueFormat.number:
-        return this.formatNumberValue(<number>value);
-      // TODO: Manage more value formats
+        return this.formatNumberDisplayValue(<number>value);
+      case DynamicFacetRangeValueFormat.date:
+        return this.formatDateDisplayValue(value);
       default:
         return `${value}`;
     }
   }
 
-  private static formatNumberValue(value: number): string {
+  private static formatNumberDisplayValue(value: number): string {
     const numberOfDecimals = NumberUtils.countDecimals(value);
     return Globalize.format(value, `n${numberOfDecimals}`);
   }
 
-  public static formatRangeValue(range: IRangeValue, valueFormat: DynamicFacetRangeValueFormat, valueSeparator: string) {
-    const formattedStart = this.formatValue(range.start, valueFormat);
-    const formattedEnd = this.formatValue(range.end, valueFormat);
+  private static formatDateDisplayValue(value: RangeType): string {
```

Comment:
Instead of tacking on stuff on a generic "Util" class, I think it would have been better to create a "Range" class with "parse()" or "format()" method.

I think it would be much more expressive.

Otherwise I fear this will become a "put code in here if you don't know where to put it" kind of thing ;)